style: add ESLint to the project

rohitgeddam opened the pull request:
add eslint to the project to enforce good coding style and catch
errors early. Following airbnb style guide.

Fixes: #155

rohitgeddam committed:
style(): add ESLint to the project

add eslint to the project to enforce good coding style and catch
errors early. Following airbnb style guide.

Fixes: #155

rohitgeddam commented:
If you need me to add specific rules in eslint config please let me know.

LalitNM commented on package.json:
```diff
@@ -6,7 +6,8 @@
     "test:watch": "mocha --watch ./tests/**/*.test.js",
-    "test": "mocha ./tests/**/*.test.js"
+    "test": "mocha ./tests/**/*.test.js",
+    "lint": "eslint"
```
You missed something here.

sesamechicken replied:
@LalitNM What was missed? Code reviews should be as specific as possible 😃

rohitgeddam replied:
@LalitNM can you please point out what I missed

sesamechicken replied:
The only thing I see that you could add would be a dir for eslint to look. Also, I find Airbnb's rules a little restrictive and overly-aggressive in some areas. I don't know who to the official owner of this project is, but linting rules are opinionated and they should chime in.

As for your PR, for the job to run, you want to tell it where to look, so line 10 would look like this:
`"lint": "eslint ./public/**/*.js ./serve/**/*.js"`

rohitgeddam replied:
Should I change style guide from airbnb to standard one.

LalitNM replied:
> Should I change style guide from airbnb to standard one.

Leave it to Airbnb's rules are fine.

LalitNM replied:
> Should I change style guide from airbnb to standard one.

Leave it to Airbnb.

LalitNM reviewed (commented):
Let's wait for @sesamechicken's review, meanwhile you can update contributing guidelines.

rohitgeddam committed:
modified .eslintrc.json

remove airbnb style enforcement
add directory for eslint to check in package.json

Fixes #155

LalitNM commented:
@rohitgeddam I missed a few changes, listing them here.
1. Add tests folder for linting, use `.js` as extension.
2. Add main `app.js` for linting.
3. Move the content of `eslintrc.json` to `package.json`(Do not add an extra file).

@rohitgeddam just add one line in the [CONTRIBUTING.md](https://github.com/KamandPrompt/baat-cheet/blob/master/CONTRIBUTING.md) in the section 'Submitting a Pull Request' about running `npm run lint`. Before committing the changes and tell the contributor to fix all the errors before commiting.

sesamechicken commented:
> 3. Move the content of `eslintrc.json` to `package.json`(Do not add an extra file).

Why? This is a great separation of concerns and keeps the `package.json` from growing out of control, because you're going to have unit test settings, as well.

LalitNM commented:
> This is a great separation of concerns and keeps the package.json from growing out of control

Fair enough.

sesamechicken commented:
Where is this at?

LalitNM commented:
> add one line in the CONTRIBUTING.md in the section 'Submitting a Pull Request' about running `npm run lint`.

I thought about this and reached to conclusion that we should do this after applying linting in all files.

LalitNM committed:
added all files for linting in package.json

LalitNM commented:
Merging this, will open another issue for applying linting changes in all files.
